tests/rbd/test_rbd_live_migrate_encryption.py

```diff
@@ -0,0 +1,539 @@
+    map_config = {
+        "pool": target_pool,
+        "image": target_image,
+        "device-type": config.get("device_type", "nbd"),
+    }
+    options = create_map_options(kw[pool]["encryption_config"])
+    map_config.update(
+        {
+            "options": options,
+        }
+    )
+    out, err = rbd.device.map(**map_config)
+
+    if err:
+        log.error(f"Failed to map migrated image {target_pool}/{target_image}")
+        return 1
+    else:
+        log.info(f"Successfully mapped the migrated image {target_pool}/{target_image}")
+
+    out, err = rbd.device.unmap(**map_config)
+    if err:
+        log.error(f"Failed to unmap migrated image {target_pool}/{target_image}")
+        return 1
+    else:
+        log.info(
+            f"Successfully unmapped the migrated image {target_pool}/{target_image}"
+        )
+
+    data_integrity_spec = {
+        "first": {
+            "image_spec": f"{pool}/{image}",
+            "rbd": rbd,
+            "client": kw["client"],
+            "file_path": f"/tmp/{random_string(len=3)}",
+        },
+        "second": {
+            "image_spec": f"{target_pool}/{target_image}",
+            "rbd": rbd,
+            "client": kw["client"],
+            "file_path": f"/tmp/{random_string(len=3)}",
+        },
+    }
+
+    rc = check_data_integrity(**data_integrity_spec)
+    if rc:
+        log.error(f"Data consistency check failed for {target_pool}/{target_image}")
+        return 1
+    else:
+        log.info("Data is consistent between the source and target images.")
+
+
+def run(**kw):
+    """
+    This test verifies Live migration of qcow2 format images with encryption
+    Args:
+        kw: test data
+    Returns:
+        int: The return value. 0 for success, 1 otherwise
+
+    """
+    try:
+
+        log.info(
+            f"{kw['config']['testconfig']['id']} - {kw['config']['testconfig']['name']}"
+        )
+
+        if kw.get("client_node"):
+            client = get_node_by_id(kw.get("ceph_cluster"), kw.get("client_node"))
+        else:
+            client = kw.get("ceph_cluster").get_nodes(role="client")[0]
+        rbd_obj = initial_rbd_config(**kw)
+        pool_types = rbd_obj.get("pool_types")
+        kw.update({"cleanup_files": []})
+        if kw["config"]["testconfig"]["id"] == "CEPH-83596444":
+            func = migration_encrypted_raw_images
+
+        elif kw["config"]["testconfig"]["id"] == "CEPH-83596588":
+            func = migration_encrypted_qcow_images
+
+        if rbd_obj:
+            log.info("Executing test on Replicated and EC pool")
+            if func(rbd_obj, client, **kw):
+                return 1
+            log.info(f"Test {kw['config']['testconfig']['name']} is successful")
+
+    except Exception as e:
+        log.error(f"Test {kw['config']['testconfig']['name']} failed: {str(e)}")
+        return 1
+
+    finally:
+        for file in kw["cleanup_files"]:
+            out, err = client.exec_command(sudo=True, cmd=f"rm -f {file}")
```

Comment:
If client.exec_command fails to delete a file or cleanup fails to delete a pool or image, no error is logged, and the function continues. This might leave:

- Temporary files in /tmp, consuming disk space.
- Pools or images in the Ceph cluster, leading to resource wastage.
- Inconsistent states in the system that complicate subsequent tests.

so we can have one try block for deleting temporary files and another try block for resource cleanup. WDYT?